containers/init/clone/clone.sh

```diff
@@ -16,9 +16,23 @@
 set -ex
 cd /src/workspace
 ls -A | xargs -r rm -fr
+
+# The git clone command clones code off the master branch by default or a branch
```

Comment:
I would rewrite it to focus on the problem that is being solved:

The git clone command clones code off the master branch by default or a branch or tag specified through the flag --branch. However, the flag --branch is not supported for commits to GitHub repositories, so the default would be to (unnecessarily) clone the master branch and then checkout the desired branch.

The process below initializes an empty git repository, adds and fetches objects from $CLONE_REPO, checks out $CLONE_GIT_REF and then updates the submodules. This process works around the need to clone an unnecessary branch.